### PR TITLE
Enable Auto-incrementing Transform job name in SageMakerTransformOperator

### LINKED ISSUE #21941: Sagemaker Transform Job fails if there are job with Same name
### Description

Sagemaker Transform Job fails if there are job with Same name exist. Let say I create a job name as 'transform-2021-01-01T00-30-00'  . So if I clear the airflow task run id for this so that the operator re-triggers then the Sagemaker Job creation fails because job with same name exists. So can we add  'action_if_job_exists flag where Behaviour if the job name already exists. Possible options are "increment" (default) and "fail".'

### Use case/motivation

Now in production environment failures are inevitable and with Sagemaker Jobs we have to ensure there is unique name for each run of the Job. So like the Sagemaker Processing Job operator or training operator we have an option to increment a job name by appending the count like if I run same job twice the job name will be 'transform-2021-01-01T00-30-00-1' where 1 is appended at end with the help of 'action_if_job_exists ([str](https://docs.python.org/3/library/stdtypes.html#str)) -- Behaviour if the job name already exists. Possible options are "increment" (default) and "fail".'

I have faced this issue personally on one of the task I am working on and think will save time and cost instead of running entire workflow again to get unique job names if there are other dependent task in the job by just clearing failed task id post fixing the failure in Sagemaker code , docker image input etc and that will just continue from where it failed

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by celeriev**
Add an option to check if transform job name already exists in SageMakerTransformOperator and increment it if it happen

closes: #21941

### CODE REVIEW

**Comment by potiuk** [comment]
Rebased to account for Werkzeug issue

**Comment by potiuk** [comment]
@ferruzzi @vincbeck @o-nikolas ? Can you take a look please?

File: airflow/providers/amazon/aws/operators/sagemaker.py [hunk 1]
```diff
@@ -419,6 +426,8 @@ def __init__(
         self.wait_for_completion = wait_for_completion
         self.check_interval = check_interval
         self.max_ingestion_time = max_ingestion_time
+        self.check_if_job_exists = check_if_job_exists
+        self.action_if_job_exists = action_if_job_exists
```
Comment (ferruzzi) [inline airflow/providers/amazon/aws/operators/sagemaker.py]: Add a check here to make sure the value is something you support. See the same feature [on L658](https://github.com/apache/airflow/pull/25263/files#diff-78df59ac1d8db4b424ba03ab52a1215e1f85dfd41a4f436c3d60bd1bcd4c4273R658-R664)

**Review by ferruzzi — CHANGES_REQUESTED** [review]
Looks like basically a copy/paste of the same feature in the SageMakerTrainingOperator, yeah?

**Comment by celeriev** [comment]
> Looks like basically a copy/paste of the same feature in the SageMakerTrainingOperator, yeah?

@ferruzzi yes the logic is the same. I'll make a suggestion with a generic method for this inside SageMakerBaseOperator

**Comment by ferruzzi** [comment]
Yup, cool.  Once you make that change I suggested, I'm good to go.

**Comment by celeriev** [comment]
@ferruzzi are you ok with the changes ? I added your suggestion :+1: thanks again for the review

**Comment by ferruzzi** [comment]
Yeah, I'm good to go, but I am not a committer so you'll need to catch their attention to gt this merged in.  

@potiuk ?

**Review by potiuk — APPROVED** [review]

**Comment by ferruzzi** [comment]
Thanks!

**Pull request merged**